# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/gnome2/devilspie/wnck-fix.patch
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./src/devilspie-action-resize.gob.orig	2005-03-11 10:16:42.000000000 -0500
+++ ./src/devilspie-action-resize.gob	2005-03-11 10:15:27.000000000 -0500
@@ -45,6 +45,7 @@
 
   override (DevilsPie:Action) gboolean run(DevilsPie:Action *self (check null type), Wnck:Window *window (check null type)) {
     DevilsPieActionResize *a = (DevilsPieActionResize*)self;
+    guint32 timestamp;
     /* Assume that a true/false tristate in a boolean context is correct */
     /* Assume the user is not totally stupid and has not set both
     maximize and minimize */
@@ -72,7 +71,7 @@
     if (a->minimized == TRI_TRUE)
       wnck_window_minimize(window);
     else if (a->minimized == TRI_FALSE)
-      wnck_window_unminimize(window);
+      wnck_window_unminimize(window, timestamp);
     /* Handle full screen */
     if (a->fullscreen != TRI_UNSET) {
         g_print("fullscreen\n");
